hubspot3/tickets.py

```diff
@@ -78,3 +90,185 @@ def get_all(self, limit: int = -1, **options) -> list:
+                "objects/tickets/batch-read",
+                method="POST",
+                doseq=True,
+                params=params,
+                data={"ids": partial_ids}
+            )
+            batch.update(partial_batch)
+        return batch
+
+    def get_batch(self, ids: List[int],
+                  extra_properties: Union[List[str], str] = None) -> List[dict]:
+        """given a batch of ticket, get more of their info"""
+        batch = self._get_batch(ids, extra_properties=extra_properties, with_history=False)
+        return [prettify(batch[ticket_id], id_key="objectId") for ticket_id in batch]
+
+    def get_batch_with_history(self, ids: List[int],
+                               extra_properties: Union[List[str], str] = None) -> Dict[str, dict]:
+        """given a batch of ticket, get more of their info with history"""
+        batch = self._get_batch(ids, extra_properties=extra_properties, with_history=True)
+        return batch
+
+    def _get_recent(
+        self,
+        recency_type: str,
+        limit: int = -1,
+        ticket_id_offset: int = 0,
+        time_offset: int = 0,
+        max_time_diff_ms: int = 60000,
+        **options
+    ) -> List:
+
+        limited = limit > 0
+        total_tickets = 0
+        finished = False
+        while not finished:
+            params = {}
+            if ticket_id_offset:
+                params["objectId"] = ticket_id_offset
+            if time_offset:
+                params["timestamp"] = int(time_offset)
+            params["changeType"] = recency_type
+            changes = self._call(
+                "change-log/tickets",
+                method="GET",
+                params=params,
+                doseq=True,
+                **options
+            )
+            total_tickets += len(changes)
+            finished = (len(changes) == 0) or (limited and total_tickets >= limit)
+            if len(changes) > 0:
+                ticket_id_offset = changes[-1]["objectId"]
+                time_offset = changes[-1]["timestamp"]
+
+                ids = set([change["objectId"] for change in changes])
+                properties = []
+                for change in changes:
+                    if change["changeType"] != TicketsClient.Recency.DELETED:
+                        properties += change["changes"]["changedProperties"]
+                properties = set(properties)
+
+                # This is an getting all the changed variables for all tickets,
+                # this is more data than needed, should eventually break in
+                # groups to get less discarted data
+                tickets_history = self.get_batch_with_history(list(ids),
+                                                              extra_properties=list(properties))
+
+                changes = self._merge_changes_with_history(changes, tickets_history,
+                                                           max_time_diff_ms)
+                yield changes
```

Comment:
Here you are yield a generator of lists. It can be a bit weird to parse it? As in, it looks like you're yielding pagination and the client is forced to process page-by-page. An alternative is to use the `yield from` statement, this way the caller receives one change per iteration and the request-pagination is transparent.